### PR TITLE
maksphp-6: RainTpl added, Comments, Attachments

### PR DESCRIPTION
**Pull request by miakobchuk**

### CODE REVIEW

File: core/controllers/account.php [hunk 1]
```diff
@@ -7,22 +7,34 @@ class Account extends Controller
         'http://www.maksphp.com/index.php?controller=account&action=listAll' => 'List All Tasks',
```
Comment (dmytrunets) [inline core/controllers/account.php]: do not hardcode url. Extract class UrlBuilder with functionality generate url

File: core/controllers/account.php [hunk 2]
```diff
@@ -7,22 +7,34 @@ class Account extends Controller
...
-    //TASK
-    //add posibility to view grid -> open item from grid with comments
+
+    protected $uploadsFolder = 'uploads/';
 
     public function index()
     {
+        $this->templateInstance->assign('sub_block', 'account_index');
+        $this->templateInstance->assign('title', 'Account index');
+        $data = 'Hello, ' . $_SESSION['user'] . ' we glad to see you here';
```
Comment (dmytrunets) [inline core/controllers/account.php]: what if key "user"  not exists in $_SESSION?

File: core/controllers/account.php [hunk 3]
```diff
@@ -7,22 +7,34 @@ class Account extends Controller
...
         $getAllUsersQuery = 'SELECT * FROM tm_task';
-        $tasks = $this->db->executeQuery($getAllUsersQuery);
+        $items = $this->db->executeQuery($getAllUsersQuery);
 
-        $this->templateInstance->assign('sub_block', 'templates/items');
+        $this->templateInstance->assign('sub_block', 'account_items');
         $this->templateInstance->assign('title', 'All items');
-        $this->templateInstance->assign('tasks', $tasks);
+
+        if (empty($items)) {
```
Comment (dmytrunets) [inline core/controllers/account.php]: this logic can move in template

File: core/controllers/account.php [hunk 4]
```diff
@@ -30,16 +42,34 @@ public function listAll()
...
+        }
 
-        $this->templateInstance->assign('sub_block', 'templates/item');
+        $this->templateInstance->assign('sub_block', 'account_item');
         $this->templateInstance->assign('title', 'View Item');
-        $this->templateInstance->assign('task', $task);
-        $this->templateInstance->assign('comments', $comments);
+        $this->templateInstance->assign('item', $item);
+        $this->templateInstance->assign('itemId', $itemId);
+//        $this->templateInstance->assign('comments', $comments);
```
Comment (dmytrunets) [inline core/controllers/account.php]: remove

File: core/controllers/account.php [hunk 5]
```diff
@@ -47,28 +77,132 @@ public function viewItem()
...
+
+    }
+
+    public function addComment()
+    {
+        $itemId = $_GET['itemId'];
+
+        if (isset($_POST['addCommentSubmit'])) {
+            $user = $this->getUserId();
+            $insertCommentQuery = 'INSERT INTO `tm_comment` (`content`, `date`, `author_id`, `task_id`) VALUES
```
Comment (dmytrunets) [inline core/controllers/account.php]: This is buisness logic, need to extract class.

File: core/controllers/account.php [hunk 6]
```diff
@@ -47,28 +77,132 @@ public function viewItem()
...
+            return ($comments);
+        } else return 'No comments for this task';
+    }
+
+    public function prepareComments($comments)
+    {
+        $preparedComments = array();
+
+        foreach ($comments as $comment) {
+            $getCommentAuthorQuery = 'SELECT name FROM tm_user WHERE id = ' . $comment['author_id'];
```
Comment (dmytrunets) [inline core/controllers/account.php]: Read about sql injection and protection this vulnarability

File: core/controllers/account.php [hunk 7]
```diff
@@ -47,28 +77,132 @@ public function viewItem()
...
         $comments = $this->db->executeQuery($getCommentsQuery);
 
-        if(!empty($comments)) {
-            return($comments);
+        if (!empty($comments)) {
+            return ($comments);
+        } else return 'No comments for this task';
+    }
+
+    public function prepareComments($comments)
```
Comment (dmytrunets) [inline core/controllers/account.php]: is this method sould be public?

File: core/controllers/account.php [hunk 8]
```diff
@@ -47,28 +77,132 @@ public function viewItem()
...
+        return $result[0]['id'];
+    }
+
+    public function addAttachment()
+    {
+        $itemId = $_GET['itemId'];
+
+        if (isset($_POST['addAttachmentSubmit'])) {
+
+            $destination = $this->uploadsFolder . basename($_FILES['file']['name']);
```
Comment (dmytrunets) [inline core/controllers/account.php]: Is is buisiniss logic need to extract class AttachmentService

File: templates/account_index.html [hunk 9]
```diff
@@ -0,0 +1,9 @@
+Page Title: {$title}
+
+<table border="1" width="100%" bgcolor="#f5f5dc">
```
Comment (dmytrunets) [inline templates/account_index.html]: use css

File: templates/account_item_comments.html [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<p>Comments: </p>
```
Comment (dmytrunets) [inline templates/account_item_comments.html]: Next task: make ability reply on comment and show comment tree. (Read about recursion)

File: core/controllers/account.php (hunk 4 above)
Comment (miakobchuk) [inline core/controllers/account.php]: You mean that i should create separate method like:

$params = array(
            'items' => $items,
            'sub_block' => 'account_items',
            'title' => 'All items',
            'subMenuLinks' => $this->subMenuLinks,
        );
        $this->setParams($params);

protected function setParams($params) {
        foreach($params as $key => $value) {
            $this->templateInstance->assign($key, $value);
        }
    }

File: core/controllers/account.php (hunk 1 above)
Comment (miakobchuk) [inline core/controllers/account.php]: Please look at code below. 
Is it a right variant?

protected $subMenuLinks = array(
        'List All Tasks' => 'listAll',
        'Tasks Assigned to User' => 'tasksAssigned'
    );

    protected function urlBuilder($links) {
        $callers=debug_backtrace();
        $controller = strtolower($callers[1]['class']);
        foreach($links as $key => $value) {
            $linksArray[$key] = $_SERVER['HTTP_HOST'] . '/index.php?controller=' . $controller . '&action=' . $value;
        }

        return $linksArray;
    }

$linksArray = $this->urlBuilder($this->subMenuLinks);

**Pull request merged**